dids/src/main/kotlin/web5/sdk/dids/DidDht.kt

```diff
@@ -0,0 +1,424 @@
+  public val publish: Boolean? = false,
+) : CreateDidOptions
+
+/**
+ * Provides a specific implementation for creating and resolving "did:dht" method Decentralized Identifiers (DIDs).
+ *
+ * A "did:dht" DID is a special type of DID that is based on an identity key, but can be extended to contain other
+ * keys, services, and other DID Document properties. It relies upon Distributed Hash Table (DHT) provided by the
+ * BitTorrent network, and an intermediary layer called Pkarr (Public Key Addressable Records) to store and retrieve
+ * the DID Document.
+ *
+ * @property uri The URI of the "did:dht" which conforms to the DID standard.
+ * @property keyManager A [KeyManager] instance utilized to manage the cryptographic keys associated with the DID.
+ * @property didDocument The [DIDDocument] associated with the DID, created by the class.
+ */
+public class DidDht(uri: String, keyManager: KeyManager, public val didDocument: DIDDocument? = null) : Did(
+  uri,
+  keyManager
+) {
+
+  /**
+   * Resolves the current instance's [uri] to a [DidResolutionResult], which contains the DID Document
+   * and possible related metadata.
+   *
+   * @return A [DidResolutionResult] instance containing the DID Document and related context.
+   *
+   * @throws IllegalArgumentException if the provided DID does not conform to the "did:dht" method.
+   */
+  public fun resolve(): DidResolutionResult {
+    return resolve(this.uri)
+  }
+
+  public companion object : DidMethod<DidDht, CreateDidDhtOptions> {
+    override val methodName: String = "dht"
+
+    public const val TTL: Long = 7200
+
+    /**
+     * Creates a new "did:dht" DID, derived from an initial identity key, and stores the associated private key in the
+     * provided [KeyManager].
+     *
+     * The method-specific identifier of a "did:dht" DID is a z-base-32 encoded public key.
+     *
+     * **Note**: By default, no additional keys or services are added to the document
+     *
+     * @param keyManager A [KeyManager] instance where the new key will be stored.
+     * @param options Optional parameters ([CreateDidDhtOptions]) to specify additional keys, services, and optional
+     * publishing during creation.
+     * @return A [DidDht] instance representing the newly created "did:dht" DID.
+     */
+    override fun create(keyManager: KeyManager, options: CreateDidDhtOptions?): DidDht {
+      // TODO(gabe): enforce that provided keys are of supported types according to the did:dht spec
+      val opts = options ?: CreateDidDhtOptions()
+
+      // create identity key
+      val keyAlias = keyManager.generatePrivateKey(JWSAlgorithm.EdDSA, Curve.Ed25519)
+      val publicKey = keyManager.getPublicKey(keyAlias)
+
+      // build DID Document
+      val id = getDidIdentifier(publicKey)
+
+      // add identity key to relationships map
+      val identityVerificationMethod =
+        VerificationMethod.builder()
+          .id(URI.create("$id#0"))
```

Comment:
thoughts on using jwk thumbprint?

Reply:
intentionally want to add a consistent first key to the DID document, thumbprint can be used for other keys in the doc

Reply:
is the rationale behind why you want to do this documented somewhere? asking bc i know its going to come up over and over. would like to have a doc to point to

Reply:
no rationale (I can add) but it is specified here https://tbd54566975.github.io/did-dht-method/#create

the tl;dr on the rationale is:
- Have a guaranteed key to use (since it's the identity key, we can rely on it)
- Reduce size of the packet since 0 is shorter than a thumbprint

Reply:
https://github.com/TBD54566975/did-dht-method/issues/27#issuecomment-1795941823

Reply:
awesome thanks!